### PR TITLE
Improvements to LABEL and COUNT

### LINKED ISSUE #63: Improvements to LABEL and COUNT
1) LABEL should be able to change the label of Buttons and Basic Widgets

Reason: The LABEL function should allow to change the label of any widgets that have text labels, which includes Buttons and Basic Widgets.

2) COUNT should have a holder parameter (in addition of the collection parameter). As in SELECT, piles should be resolved to their content. Question is: Should we also ignore deck when counting or should this be considered by the game designer who should know what's in the holder? Because besides a deck, a holder could contain buttons, labels, etc., so the onyl widget that we would know to ignore for certain is piles.

Reason: This would allow to count the children of a holder without making a prior selection and makes COUNT more consistent with other functions that allow to operate on a holder.

### PR DESCRIPTION
**Pull request by larkob**
LABEL also works on collections and can also change the label of Buttons and Basic Widgets

COUNT can also use a `holder` parameter to count any holder's direct children (matching `dropTarget` and resolving piles).

It initially also counted any other widgets it finds in the holder, but as that would make it inconsistent with other button operations, I've removed that part.

It also contains simple fixes to json editor.

Resolves #63

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3272/pr-test (or any other room on that server)

File: client/js/widgets/widget.js [hunk 1]
```diff
@@ -357,6 +357,21 @@ export class Widget extends StateManaged {
...
+    if (this.p('text') !== undefined) {
+      if(mode == 'inc' || mode == 'dec')
+        this.p('text', (parseInt(this.p('text')) || 0) + (mode == 'dec' ? -1 : 1) * text);
+      else if(Array.isArray(text))
+        this.p('text', text.join(', '));
+      else if(typeof text == 'string' && text.match(/^[-+]?[0-9]+(\.[0-9]+)?$/))
+        this.p('text', +text);
+      else
+        this.p('text', text);
+      if (debug) console.log(`changing ${this.id} ${mode} ${text}`)
```
Comment (ArnoldSmith86) [inline client/js/widgets/widget.js:370]: Not sure where this debug message comes from but it seems unnecessary to me and weird that it gets toggled by the `debug` property.

File: client/js/jsonedit.js [hunk 2]
```diff
@@ -383,11 +383,13 @@ function jeAddCommands() {
 
   jeAddButtonOperationCommands('CLICK', { collection: 'DEFAULT', count: 1 });
   jeAddButtonOperationCommands('COMPUTE', { operation: '+', operand1: 1, operand2: 1, operand3: 1, variable: 'COMPUTE' });
-  jeAddButtonOperationCommands('COUNT', { collection: 'DEFAULT', variable: 'COUNT' });
-  jeAddButtonOperationCommands('FLIP', { count: 0, face: null });
+  jeAddButtonOperationCommands('COUNT', { collection: 'DEFAULT', holder: null, variable: 'COUNT' });
+  jeAddButtonOperationCommands('CLONE', { source: 'DEFAULT', collection: 'DEFAULT', xOffset: 0, yOffset: 0, count: 1 });
```
Comment (ArnoldSmith86) [inline client/js/jsonedit.js:387]: This is missing `properties`, right?

File: client/js/widgets/button.js [hunk 3]
```diff
@@ -376,13 +380,22 @@ export class Button extends Widget {
...
       if(a.func == 'LABEL') {
-        setDefaults(a, { value: 0, mode: 'set' });
+        setDefaults(a, { value: 0, mode: 'set', collection: 'DEFAULT' });
         if([ 'set', 'dec', 'inc' ].indexOf(a.mode) == -1)
           problems.push(`Warning: Mode ${a.mode} will be interpreted as add.`);
-        this.w(a.label, label=> {
-          if (this.p('debug')) console.log(`changing ${a.label} ${a.mode} ${a.value}`)
-          label.setText(a.value, a.mode)
-        });
+        if(a.label !== undefined) a.widget = a.label; // remove this line to deprecate the 'label' property in favor of 'widget'
```
Comment (ArnoldSmith86) [inline client/js/widgets/button.js:386]: Didn't we say no aliases? Shouldn't this do a file version bump instead? (We probably shouldn't wait for the pile rewrite anymore.)

**Review by ArnoldSmith86 — APPROVED** [review]
I made a few comments but I don't mind if it gets merged as is.

File: client/js/widgets/button.js (hunk 3 above)
Comment (larkob) [reply client/js/widgets/button.js]: Right, I was intending to prepare a file version bump, but I'll better do that as a dedicated PR and just get this PR merged.

**Pull request merged**